src/middleware/thunk.js

```diff
@@ -1,10 +1,6 @@
-export default function thunkMiddleware(getState) {
-  return (next) => {
-    const recurse = (action) =>
-      typeof action === 'function' ?
-        action(recurse, getState) :
-        next(action);
-
-    return recurse;
-  };
+export default function thunkMiddleware({ dispatch, getState }) {
+  return (next) => (action) =>
+    typeof action === 'function' ?
+      action(dispatch, getState) :
```

Comment:
For example, the action creator might want to exit early if some data is already available in the state. Also useful for dispatching an action and then reading the result (e.g. authentication workflow).